crates/ruff_python_formatter/src/statement/stmt_import_from.rs

```diff
@@ -1,12 +1,49 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
-use ruff_formatter::{write, Buffer, FormatResult};
+use crate::builders::{optional_parentheses, PyFormatterExtensions};
+use crate::{AsFormat, FormatNodeRule, PyFormatter};
+use ruff_formatter::prelude::{dynamic_text, format_with, space, text};
+use ruff_formatter::{write, Buffer, Format, FormatResult};
 use rustpython_parser::ast::StmtImportFrom;
+use std::iter;
 
 #[derive(Default)]
 pub struct FormatStmtImportFrom;
 
 impl FormatNodeRule<StmtImportFrom> for FormatStmtImportFrom {
     fn fmt_fields(&self, item: &StmtImportFrom, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtImportFrom {
+            module,
+            names,
+            range: _,
+            level,
+        } = item;
+
+        let level_str = level
+            .map(|level| ".".repeat(level.to_usize()).collect())
+            .unwrap_or(String::default());
+
+        write!(
+            f,
+            [
+                text("from"),
+                space(),
+                dynamic_text(&level_str, None),
+                module.as_ref().map(AsFormat::format),
```

Comment:
i was surprised the wasn't a blanket `Option<T>: T: AsFormat` impl for `.format()`

Reply:
I think I ran into issues when I tried adding it last time because Rust doesn't know that T doesn't implement Format and AsFormat at the same time